tempesta_fw/connection.h

```diff
@@ -96,6 +102,38 @@ typedef struct {
 	TfwMsg * (*conn_msg_alloc)(TfwConnection *conn);
 } TfwConnHooks;
 
+static inline void
+tfw_connection_get(TfwConnection *conn)
+{
+	atomic_inc(&conn->refcnt);
+}
+
+static inline bool
+tfw_connection_put(TfwConnection *conn)
+{
+	if (unlikely(!conn))
+		return false;
+	if (likely(atomic_read(&conn->refcnt) == 1))
+		smp_rmb();
```

Comment:
Do we need the barrier on x86-64 which we only support?
I created #241 for the compile-time assertion.